en/references/tested-judge/template-description/README.md

```diff
@@ -0,0 +1,392 @@
+The only limitation for the HTML code snippets, is that they not may contain `code` tags inside, even in strings.
+
+### Examples
+A first example is a Markdown template with a Kotlin instance.
+A second example is a HTML template with a Haskell instance.
+
+#### Markdown
+Template:
+````mako
+```tested
+> heir(8, 10)
+[10, 4, 15, 11, 7, 5, 3,
+ 2, 16, 12, 1, 6, 13, 9, 14, 8]
+```
+````
+
+Instance Kotlin:
+````markdown
+```console?lang=kotlin&prompt=>
+> heir(8, 10)
+listOf(10, 4, 15, 11, 7, 5, 3, 2, 16, 12, 1, 6, 13, 9, 14, 8)
+```
+````
+
+#### HTML
+Template:
+```mako
+<div class="highlighter-rouge language-${language}">
+<pre class="highlight"><code class="color tested code" id="code">\
+> encode("And now for something completely different.",
+         1)
+"And now for something completely different."\
+</code></pre>
+</div>
+```
+
+Instance Haskell:
+```html
+<div class="highlighter-rouge language-haskell">
+<pre class="highlight"><code class="color tested code" id="code">&gt; <span class="nf">encode</span> <span class="p">(</span><span class="s">&quot;And now for something completely different.&quot;</span><span class="p">)</span> <span class="p">(</span><span class="mi">1</span> <span class="ow">::</span> <span class="kt">Int</span><span class="p">)</span>
+<span class="s">&quot;And now for something completely different.&quot;</span></code></pre>
+</div>
+```
+
+## Language specific information
+The Mako-templates have also conditional constructs,
+this makes it possible to add information for only one specific programming language.
+
+### Example
+We will look to an example in Markdown for Java and C, but a similar structure could be used in HTML.
+
+Template:
+```mako
+Common text
+
+% if language == 'java':
+The function must be static declared in the class ${namespace}.
+% endif
+```
+
+Instantie Java:
+```markdown
+Common text
+
+The function must be static declared in the class Solution.
+```
+
+Instantie C:
+```markdown
+Common text
+```
+
+## Custom variables
+When you use a name frequent in the text, you can define a variable in Make,
+the advantages is avoiding multiple function calls that returns the same value.
+
+### Example
+We will now look at an example in Markdown without instances.
+```mako
+<% list = natural_type_name("list") %>\
+Write a function ${function_name("heir")}, with has two parameters $k$ en $n$,
+which have both the type ${type_name("integer")}, you may assume that $k >= 2$.
+The function must return a ${list} of the type ${type_name(("list", "integer"))}
+that contains the order of the removed children of the circle.
+The first removed child must be at the first position in the ${list},
+and the eventuual heir must last in the ${list}.
+Use the numbers which are used to order the children in the ${list},
+as elements in the ${list}.
+```
+
+## Complete HTML template
+Below follows a complete HTML template description.
+```mako
+<p>
```

Comment:
Is this useful? Especially the console prompt is basically unreadable.

Reply:
I think it's useful to indicate that the prompt will be html encoded.